### PR TITLE
Headers refactor

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/header.rs [hunk 1]
```diff
@@ -12,6 +12,199 @@
...
+                self.insert(key, value.to_string().into_header_value());
+            }
+        }
+    }
+
+    pub fn get<T: IntoHeaderName>(&self, name: T) -> Option<&HeaderValue> {
+        self.headers.get(&name.into_header_name())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
```
Comment (caspervonb) [inline async-nats/src/header.rs:76]: Should serialization/deserialization (from_str) be here, or with the rest of protocol stuff?

**Review by caspervonb — CHANGES_REQUESTED** [review]

File: async-nats/src/header.rs (hunk 1 above)
Comment (Jarema) [reply async-nats/src/header.rs:76]: That's a good question I tried to answer myself.

It feels like it belongs to protocol stuff, but at the same time, header serialisation is quite specific to headers only, especially NATS headers.

**Review by caspervonb — APPROVED** [review]
lgtm

File: async-nats/src/header.rs (hunk 1 above)
Comment (caspervonb) [reply async-nats/src/header.rs]: Alright I'll revisit this next week or so. Lets merge and get KV out.

**Pull request merged**